File: SpringBootJpa/src/main/java/com/example/java/simple/repository/SimpleCustomJpqlRepository.java
package com.example.java.simple.repository;

import com.example.java.simple.domain.Customer;

import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import java.util.List;

public class SimpleCustomJpqlRepository implements CustomJpqlRepository {

    @PersistenceContext
    private EntityManager em;

    @Override
    public List<Customer> findByName(String name) {
        return em.createNamedQuery("Custom.findByName", Customer.class)
                .setParameter("name", name)
                .getResultList();
    }
}

File: SpringBootJpa/src/main/kotlin/com/example/kotlin/simple/repository/SimpleCustomJpqlRepository.kt
package com.example.kotlin.simple.repository

import com.example.kotlin.simple.domain.Customer
import jakarta.persistence.EntityManager

class SimpleCustomJpqlRepository(val em: EntityManager) : CustomJpqlRepository {

    override fun findByName(name: String): List<Customer> {
        return em.createNamedQuery("Custom.findByName", Customer::class.java)
                .setParameter("name", name)
                .resultList
    }
}
